packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -24,33 +24,87 @@ export const displayTxResult = (
   if (typeof displayContent === "bigint") {
     try {
       const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
+      if (asText) {
+        if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+          return asNumber;
+        } else {
+          return "Ξ" + formatEther(displayContent);
+        }
       } else {
-        return "Ξ" + formatEther(displayContent);
+        return <NumberDisplay value={displayContent} />;
       }
     } catch (e) {
       return "Ξ" + formatEther(displayContent);
     }
   }
 
-  if (typeof displayContent === "string" && isAddress(displayContent)) {
-    return asText ? displayContent : <Address address={displayContent} />;
+  if (typeof displayContent === "string") {
+    if (isAddress(displayContent)) {
+      return asText ? displayContent : <Address address={displayContent} />;
+    }
+
+    if (isHex(displayContent)) {
+      return displayContent; // don't add quotes
```

Comment:
I think we could remove the comment

Reply:
Personally, I would keep it, because it seems weird to just return the raw value in this specific condition.